java/arcs/core/storage/DirectStore.kt

```diff
@@ -59,7 +59,9 @@ class DirectStore<Data : CrdtData, Op : CrdtOperation, T> /* internal */ constru
     private val log = TaggedLog { "DirectStore" }
 
     /** True if this store has been closed. */
-    var closed = false
+    var closedMonitor = Any()
```

Comment:
what is the purpose of creating a separate object for lock instead of using proxyManager directly?

Reply:
I guess we could do that. In theory, `closed` doesn't have to only related to the `proxyManager`, so tying it to that object felt overly restrictive. But I can make that change.